x/interchainstaking/keeper/proposal_handler.go

```diff
@@ -16,16 +16,37 @@ import (
 	"github.com/ingenuity-build/quicksilver/x/interchainstaking/types"
 )
 
-// HandleRegisterZoneProposal is a handler for executing a passed community spend proposal.
+// HandleRegisterZoneProposal is a handler for executing a register zone proposal.
 func (k *Keeper) HandleRegisterZoneProposal(ctx sdk.Context, p *types.RegisterZoneProposal) error {
+	var (
+		baseZone types.Zone
+		found    bool
+	)
+
 	// get chain id from connection
 	chainID, err := k.GetChainID(ctx, p.ConnectionId)
 	if err != nil {
 		return fmt.Errorf("unable to obtain chain id: %w", err)
 	}
 
+	// if subzone
+	if p.SubzoneInfo != nil {
+		if chainID != p.SubzoneInfo.BaseChainID {
```

Comment:
If we don't store SubzoneInfo.BaseChainID separately (see previous comment) then we can remove this.